File: Problems/7.reverse-integer.py
# title: Reverse Integer
# timestamp: 2025-05-20 11:46:43
# problemUrl: https://leetcode.com/problems/reverse-integer/
# memory: 17.6 MB
# runtime: 39 ms

class Solution:
    def reverse(self, x: int) -> int:
        threashold = 214748364
        out = 0
        sign = -1 if x < 0 else 1
        x *= sign
        while x:
            out *= 10
            out += x % 10
            x //= 10

            if x != 0 and (out > threashold or out < -threashold):
                return 0
        
        return out * sign

File: Problems/7.reverse-integer.go
// title: Reverse Integer
// timestamp: 2024-04-24 11:53:07
// problemUrl: https://leetcode.com/problems/reverse-integer/
// memory: 2.3 MB
// runtime: 0 ms

func reverse(x int) int {
    const threashold = 214748364
    var in, out int32 = int32(x), 0
    for in != 0 {
        out *= 10
        out += in%10
        in /= 10

        if in != 0 && (out > threashold || out < -threashold) {
            return 0
        }
    }
    
    return int(out)
}